### PR TITLE
Fixed several locations where an assumption is made that a ForeignKey…

### PR DESCRIPTION
**Pull request by CodingGorilla**
…Definition.ReferenceDocumentType will not be null

Addresses #1338

### CODE REVIEW

File: src/MartenBenchmarks/BenchAgainst/StorageFeatures.cs [hunk 1]
```diff
@@ -194,6 +194,7 @@ public IEnumerable<IFeatureSchema> AllActiveFeatures(ITenant tenant)
                     return m.ForeignKeys
                         .Where(x => x.ReferenceDocumentType != m.DocumentType)
                         .Select(keyDefinition => keyDefinition.ReferenceDocumentType)
+                        .Where(referenceDocumentType => referenceDocumentType != null)
```
Comment (oskardudycz) [inline src/MartenBenchmarks/BenchAgainst/StorageFeatures.cs]: I'd suggest to place it [in the first where](https://github.com/JasperFx/marten/pull/1339/files#diff-2db68eabb5a386b9eecd111babc6e92eR195). That would be the same way as it's done [above](https://github.com/JasperFx/marten/pull/1339/files#diff-0c90b6f6e976fbf2e6c8997ccaa8cccdR281)

Comment (CodingGorilla) [reply src/MartenBenchmarks/BenchAgainst/StorageFeatures.cs]: 👍 Got it, was copying a different location where this was already done, will update ASAP.

Comment (CodingGorilla) [reply src/MartenBenchmarks/BenchAgainst/StorageFeatures.cs]: Done.

**Comment by CodingGorilla** [comment]
Don't merge this PR, I will create a new one with squashed commits to clean up my mess once I've fixed the tests.

**Comment by CodingGorilla** [comment]
@oskardudycz I think this is done, I'm not sure how to mark your "Changes requested" as complete on GitHub, but I think this is ready to go.

File: src/Marten.Testing/Bugs/bug_1338_Validate_Null_ForeignKeyDefinition_ReferenceDocumenType.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        [Fact]
+        public void UnitOfWork_GetTypeDependencies_Should_Not_Throw_With_ExternalForeignKeyDefinitions()
+        {
+            CreateExternalTableForTesting();
+
+            StoreOptions(_ =>
+                         {
+                             _.Schema.For<ClassWithExternalForeignKey>().ForeignKey(x => x.ForiegnId, _.DatabaseSchemaName, "external_table", "id");
```
Comment (mysticmind) [inline src/Marten.Testing/Bugs/bug_1338_Validate_Null_ForeignKeyDefinition_ReferenceDocumenType.cs]: Please fix typo `ForiegnId` to `ForeignId`

**Pull request merged**